src/main/kotlin/com/github/vol0n/utbotcppclion/services/Client.kt

```diff
@@ -0,0 +1,276 @@
+                        if (response.type == Testgen.ProjectConfigStatus.LINK_COMMANDS_JSON_NOT_FOUND) "link_commands.json" else "compile_commands.json"
+                    notifyError(
+                        "Project is not configured properly: $missingFileName is missing in the build folder.",
+                        project, AskServerToGenerateJsonForProjectConfiguration()
+                    )
+                }
+                else -> notifyUnknownResponse(response, project)
+            }
+        }
+
+        val request = getProjectConfigRequestMessage(project, Testgen.ConfigMode.CHECK)
+        stub.configureProject(request).handleWithProgress(
+            "Configuring project",
+            ::progressOrNull,
+            ::handleProjectConfigResponse
+        )
+    }
+
+    fun createBuildDir() {
+        fun handleBuildDirCreation(serverResponse: Testgen.ProjectConfigResponse) {
+            when (serverResponse.type) {
+                Testgen.ProjectConfigStatus.IS_OK -> {
+                    notifyInfo("Build dir was created!", project)
+                    configureProject()
+                }
+                Testgen.ProjectConfigStatus.BUILD_DIR_CREATION_FAILED -> {
+                    notifyInfo("Failed to create build dir! ${serverResponse.message}", project)
+                }
+                else -> notifyUnknownResponse(serverResponse, project)
+            }
+        }
+
+        val request = getProjectConfigRequestMessage(project, Testgen.ConfigMode.CREATE_BUILD_DIR)
+        stub.configureProject(request).handleWithProgress(
+            "Ask server to create build dir",
+            ::progressOrNull,
+            ::handleBuildDirCreation
+        )
+    }
+
+    fun generateJSon() {
+        fun handleJSONGeneration(serverResponse: Testgen.ProjectConfigResponse) {
+            when (serverResponse.type) {
+                Testgen.ProjectConfigStatus.IS_OK -> notifyInfo("Successfully configured project!", project)
+                Testgen.ProjectConfigStatus.RUN_JSON_GENERATION_FAILED -> notifyError(
+                    "UTBot tried to configure project, but failed with the " +
+                            "following message: ${serverResponse.message}", project
+                )
+                else -> notifyUnknownResponse(serverResponse, project)
+            }
+        }
+
+        val request = getProjectConfigRequestMessage(project, Testgen.ConfigMode.GENERATE_JSON_FILES)
+        stub.configureProject(request).handleWithProgress(
+            "Ask server to generate json",
+            ::progressOrNull,
+            ::handleJSONGeneration,
+        )
+    }
+
+    private suspend fun heartBeat() {
+        try {
+            stub.heartbeat(Testgen.DummyRequest.newBuilder().build())
+            connectionChangedPublisher.onChange(connectionStatus, ConnectionStatus.CONNECTED)
+            connectionStatus = ConnectionStatus.CONNECTED
+        } catch (e: Exception) {
```

Comment:
What are the possible exceptions? Please try to use more specific exception types whenever possible. And also in other cases